File: ruby/y2022/day01.rb
#!/usr/bin/env ruby
#
# Calorie Counting

require_relative '../day'

class Day01 < Day
  def part1
    sums = calorie_sums
    puts sums.max
  end

  def part2
    sums = calorie_sums

    max = sums.max
    top_three = max
    sums -= [max]

    max = sums.max
    top_three += max
    sums -= [max]

    max = sums.max
    top_three += max

    puts top_three
  end

  def calorie_sums
    lines = data_lines(1, false)
    groups = lines.slice_when { |line| line.empty? }
    groups.map { |lines| lines.map(&:to_i).sum }
  end
end

if __FILE__ == $PROGRAM_NAME
  require_relative '../aoc'

  aoc(2022, 1)
end

File: python/y2022/day01.py
# Calorie Counting

import itertools

from utils import *


def part1(env):
    grouped_lines = data_file_lines(env, preserve_blank_lines=True)
    sums = [sum(int(line) for line in nums) for nums in grouped_lines]
    print(sorted(sums)[-1])


def part2(env):
    grouped_lines = data_file_lines(env, preserve_blank_lines=True)
    sums = [sum(int(line) for line in nums) for nums in grouped_lines]
    print(sum(sorted(sums)[-3:]))
